### PR TITLE
[ResourceGraph] Add tests

### PR DESCRIPTION
**Pull request by Jack-An-14**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/resourcegraph/Azure.ResourceManager.ResourceGraph/tests/Azure.ResourceManager.ResourceGraph.Tests.csproj [hunk 1]
```diff
@@ -1,4 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
```
Comment (HarveyLink) [inline sdk/resourcegraph/Azure.ResourceManager.ResourceGraph/tests/Azure.ResourceManager.ResourceGraph.Tests.csproj:2]: Is this needed?

File: sdk/resourcegraph/Azure.ResourceManager.ResourceGraph/tests/ResourceGraphManagementTestBase.cs [hunk 2]
```diff
@@ -14,6 +15,11 @@ public class ResourceGraphManagementTestBase : ManagementRecordedTestBase<Resour
     {
         protected ArmClient Client { get; private set; }
 
+        public AzureLocation azureLocation = AzureLocation.EastUS;
+        public string DefaultRgnamePrefix = "Test";
+        public SubscriptionResource DefaultSubscription { get; private set; }
+        //public ManagementGroupResource DefaultmanagementGroupResource { get; private set; }
```
Comment (HarveyLink) [inline sdk/resourcegraph/Azure.ResourceManager.ResourceGraph/tests/ResourceGraphManagementTestBase.cs:21]: Remove un-used line

File: sdk/resourcegraph/Azure.ResourceManager.ResourceGraph/tests/Tests/ResourceGraphTest.cs [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+            Assert.NotNull(error.Errors[0].Code);
+            Assert.NotNull(error.Errors[1].Code);
+        }
+
+        [Test]
+        public async Task ResourcesMalformedQueryAsyncTest()
+        {
+            var tenantsCollection = Client.GetTenants();
+            var tenantList = await tenantsCollection.GetAllAsync().ToEnumerableAsync();
+            var item = tenantList.FirstOrDefault();
```
Comment (HarveyLink) [inline sdk/resourcegraph/Azure.ResourceManager.ResourceGraph/tests/Tests/ResourceGraphTest.cs:213]: We could make a method for these code

File: sdk/resourcegraph/Azure.ResourceManager.ResourceGraph/tests/Azure.ResourceManager.ResourceGraph.Tests.csproj (hunk 1 above)
Comment (An-JAY) [reply sdk/resourcegraph/Azure.ResourceManager.ResourceGraph/tests/Azure.ResourceManager.ResourceGraph.Tests.csproj]: done delete

File: sdk/resourcegraph/Azure.ResourceManager.ResourceGraph/tests/ResourceGraphManagementTestBase.cs (hunk 2 above)
Comment (An-JAY) [reply sdk/resourcegraph/Azure.ResourceManager.ResourceGraph/tests/ResourceGraphManagementTestBase.cs]: done

File: sdk/resourcegraph/Azure.ResourceManager.ResourceGraph/tests/Tests/ResourceGraphTest.cs [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+{
+    public class ResourceGraphTest : ResourceGraphManagementTestBase
+    {
+        private TenantResource _tenant;
+
+        public ResourceGraphTest(bool isAsync)
+            : base(isAsync)//, RecordedTestMode.Record)
+        {
+        }
+        public async Task<TenantResource> getTenantResourceAsync()
```
Comment (HarveyLink) [inline sdk/resourcegraph/Azure.ResourceManager.ResourceGraph/tests/Tests/ResourceGraphTest.cs:27]: Blank line and Upper case

File: sdk/resourcegraph/Azure.ResourceManager.ResourceGraph/tests/Tests/ResourceGraphTest.cs [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+using Azure.ResourceManager.ResourceGraph.Models;
+using Azure.ResourceManager.ResourceGraph.Tests;
+using Azure.ResourceManager.Resources;
+using NUnit.Framework;
+
+namespace Azure.Management.ResourceGraph.Tests
+{
+    public class ResourceGraphTest : ResourceGraphManagementTestBase
+    {
+        private TenantResource _tenant;
```
Comment (HarveyLink) [inline sdk/resourcegraph/Azure.ResourceManager.ResourceGraph/tests/Tests/ResourceGraphTest.cs:21]: In C#, underscore may point to `Discards`, please use normal CamelCase here

**Comment by An-JAY** [comment]
> @microsoft-github-policy-service agree [company="{wircresoft}"]

File: sdk/resourcegraph/Azure.ResourceManager.ResourceGraph/tests/Tests/ResourceGraphTest.cs [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+
+        [Test]
+        public async Task ResourcesQueryOptionsTest()
+        {
+            var queryContent = new QueryContent("project id")
+            {
+                Subscriptions = { DefaultSubscription.Data.SubscriptionId },
+                Options = new QueryRequestOptions
+                {
+                    SkipToken = "ew0KICAiJGlkIjogIjEiLA0KICAiTWF4Um93cyI6IDQsDQogICJSb3dzVG9Ta2lwIjogNCwNCiAgIkt1c3RvQ2x1c3RlclVybCI6ICJodHRwczovL2FybXRvcG9sb2d5Lmt1c3RvLndpbmRvd3MubmV0Ig0KfQ==",
```
Comment (Yao725) [inline sdk/resourcegraph/Azure.ResourceManager.ResourceGraph/tests/Tests/ResourceGraphTest.cs:122]: Is it ok to leak this token?

Comment (An-JAY) [reply sdk/resourcegraph/Azure.ResourceManager.ResourceGraph/tests/Tests/ResourceGraphTest.cs:122]: Track 1 also has it

**Review by Yao725 — APPROVED** [review]
LGTM, please also address the license check.

**Comment by An-JAY** [comment]
> 

@microsoft-github-policy-service agree

**Pull request merged**